api/globalSearch/index.ts

```diff
@@ -70,8 +70,10 @@ export function generateServices(refDataHMCTS: RefDataHMCTSService[]): HMCTSServ
   const globalSearchServices: HMCTSServiceDetails[] = [];
   globalSearchServiceIdsArray.forEach((serviceId) => {
     // search for the service name based on the globalSearchServiceId
-    const jurisdiction = refDataHMCTS?.length > 0 ? refDataHMCTS.filter((x) => x.ccd_service_name?.toLowerCase() === serviceId.toLowerCase()) : null;
-    if (jurisdiction?.length > 0) {
+    const jurisdiction = refDataHMCTS?.length > 0 ? refDataHMCTS.filter((x) => {
+      return x && (x.ccd_service_name?.toLowerCase() === serviceId.toLowerCase());
+    }) : null;
+    if (jurisdiction && jurisdiction.length > 0) {
```

Comment:
Optional chaining should be equivalent to this - you don't have to change it, but there don't seem to be any advantages in the change